subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -163,6 +398,10 @@ private DefaultGroovyJavaJointCompileSpec createSpec() {
         spec.setGroovyClasspath(Lists.newArrayList(getGroovyClasspath()));
         spec.setCompileOptions(compileOptions);
         spec.setGroovyCompileOptions(groovyCompileOptions);
+        if (getOptions().isIncremental()) {
+            spec.setCompilationMappingFile(sourceClassesMappingFile);
```

Comment:
I think we shouldn't use the same file for tracking the source mapping and for the changed information during the current compilation. In the spike, I used the different files though the same file format. That was more accidental then intended.

I think we should keep the mapping which is persistent between tasks in a persistent index cache instead of a plain text file. The compiler plugin should write a text file, since we don't have many classes on the compiler daemon to write a persistent index cache.

That reminds me that we should have test for incremental compilation when doing Groovy compilation on a worker daemon.

Reply:
This is not resolved yet, I need some time to understand persistent index cache.

Reply:
No worries, we can use a persistent indexed cached in a follow up step.